zerver/lib/outgoing_webhook.py

```diff
@@ -41,23 +41,25 @@ def process_event(self, event):
         # type: (Dict[str, Any]) -> Tuple[Dict[str ,Any], Dict[str, Any]]
         raise NotImplementedError()
 
-    # Given a successful response to the outgoing webhook REST operation, returns the message
-    # that should be sent back to the user.
+    # Given a successful response to the outgoing webhook REST operation, determines if response is required
+    # to be sent to user and in such case, returns the message that should be sent back.
     #
     # The response will be the response object obtained from REST operation.
     # The event will be the same as the input to process_command.
-    # The returned message will be a dictionary which should have "response_message" as key and response message to
-    # be sent to user as value.
+    # The returned message will be a dictionary which should have atleast
+    # 1. "response_required" as key and boolean True or False as value.
```

Comment:
I think the fact that later changes would require changes in all of the interfaces classes is a GOOD thing.  That way, if we update the return value, we can rely on the type system to help us figure out the places we need to update (if it's just a dictionary, it's really easy for us to miss a place).